### PR TITLE
Azure CLI 2.17.0

### PR DESCRIPTION
**Pull request by azclibot**
The latest release of the Azure CLI

### CODE REVIEW

File: Formula/azure-cli.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@ class AzureCli < Formula
 
   desc "Microsoft Azure CLI 2.0"
   homepage "https://docs.microsoft.com/cli/azure/overview"
-  url "https://codeload.github.com/Azure/azure-cli/legacy.tar.gz/adb59a62ecbbea5e7c7bddd5e062ef7806825d02"
-  version "2.16.0"
-  sha256 "dc8318bef61fb1054d9a64f6243b203438d8c66114c82879339aae568460ff58"
+  url "https://codeload.github.com/Azure/azure-cli/legacy.tar.gz/57ce1e6689dcbdb21f62bae6f520e74e95c0c4e0"
```
Comment (chenrui333) [inline Formula/azure-cli.rb:6]: @azclibot can we update this url to `https://github.com/Azure/azure-cli/archive/azure-cli-2.17.0.tar.gz`, then we can get rid of version line.

Comment (chenrui333) [reply Formula/azure-cli.rb:6]: Furthermore, I wonder if it would be possible to publish that big source archive to pypi and use that instead. Thanks!

Comment (Juliehzl) [reply Formula/azure-cli.rb:6]: Hi @chenrui33, thanks a lot for your feedback. 
Could we keep it now to catch product release schedule and we will make the change in next release?

Comment (carlocab) [reply Formula/azure-cli.rb:6]: Hi @Juliehzl, I think it's better to change it now, as it might be forgotten if we wait till the next release.

**Comment by fxcoudert** [comment]
Still not compatible with Python 3.9?

**Comment by chenrui333** [comment]
> Still not compatible with Python 3.9?

looks like it (https://github.com/Azure/azure-cli/issues/15465, https://github.com/Azure/azure-cli/issues/15580, https://github.com/Azure/azure-cli/issues/16318)

File: Formula/azure-cli.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/azure-cli.rb:6]: @julienXX I will log a ticket on your side (fine with automate the style for the next run) and make the change now.

**Review by chenrui333 — APPROVED** [review]

**Comment by carlocab** [comment]
Problem reported here: https://github.com/Homebrew/brew/issues/10152#issuecomment-753334171

Working on it now.

**Comment by carlocab** [comment]
Fix: https://github.com/Homebrew/homebrew-core/pull/68126

Though, a large number of the resources appear to be outdated. It would be good to switch the URL to PyPI so that updating them can be automated.

**Pull request closed without merging**